pros/src/sensors/imu.rs

```diff
@@ -0,0 +1,326 @@
+use pros_sys::{PROS_ERR, PROS_ERR_F};
+use snafu::Snafu;
+
+use crate::error::{bail_on, map_errno, PortError};
+
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq)]
+pub struct InertialStatus {
+    pub raw: pros_sys::imu_status_e_t,
+    pub calibrating: bool,
+    pub error: bool,
+}
+
+impl From<pros_sys::imu_status_e_t> for InertialStatus {
+    fn from(value: pros_sys::imu_status_e_t) -> Self {
+        Self {
+            raw: value,
+            calibrating: (value & pros_sys::E_IMU_STATUS_CALIBRATING) != 0,
+            error: (value & pros_sys::E_IMU_STATUS_ERROR) != 0,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct InertialSensor {
+    port: u8,
+}
+
+impl InertialSensor {
+    /// Create a new inertial sensor from a smart port index.
+    pub fn new(port: u8) -> Result<Self, InertialError> {
+        let sensor = Self { port };
+        sensor.rotation()?;
+        Ok(sensor)
+    }
+
+    /// Calibrate IMU.
+    /// This takes approximately 2 seconds, and is blocking until the IMU status flag is set properly.
+    pub fn calibrate(&self) -> Result<(), InertialError> {
+        unsafe {
+            bail_on!(PROS_ERR, pros_sys::imu_reset(self.port));
+        }
+        Ok(())
+    }
+
+    /// Check if the Intertial Sensor is currently calibrating.
+    pub fn is_calibrating(&self) -> Result<bool, InertialError> {
+        Ok(self.status()?.calibrating)
+    }
```

Comment:
There should also be a future for this.

Reply:
What should the API surface of this ideally look like, considering that `imu_reset` blocks by default?

Reply:
Oh, I was under the impression that imu_reset didn't block because you could poll it by getting the status of the IMU. If it is blocking then there isn't really a need for it.

Reply:
Actually, the PROS docs are giving me mixed signals on this. Let me look into this a little further.
![image](https://github.com/pros-rs/pros-rs/assets/42101043/1b7c02eb-8ac6-404c-be78-0d486e4b57ec)
![image](https://github.com/pros-rs/pros-rs/assets/42101043/8af6826f-bfdd-4cdc-b05d-94216932b47c)